biz.aQute.bndlib/src/aQute/bnd/osgi/resource/ResourceBuilder.java

```diff
@@ -462,11 +462,11 @@ public void addFragmentHost(String bsn, Attrs attrs) throws Exception {
 		StringBuilder filter = new StringBuilder();
 		filter.append("(").append(HostNamespace.HOST_NAMESPACE).append("=").append(bsn).append(")");
 
-		String v = attrs.getVersion();
+		String v = attrs.get(HostNamespace.CAPABILITY_BUNDLE_VERSION_ATTRIBUTE);
 		if (v != null && VersionRange.isOSGiVersionRange(v)) {
 			VersionRange range = VersionRange.parseOSGiVersionRange(v);
 			filter.insert(0, "(&");
-			filter.append(range.toFilter());
+			filter.append(range.toFilter().replaceAll("version", "bundle-version"));
 			filter.append(")");
```

Comment:
Lets not do a replaceAll. Instead, fix `VersionRange.toFilter()` (or overload it) to take the attribute name as a argument. See OSGi's `VerionRange.toFilterString(String)`. The then call here becomes `range.toFilter(HostNamespace.CAPABILITY_BUNDLE_VERSION_ATTRIBUTE)`.